native/src/main/java/io/ballerina/stdlib/data/xml/XmlParser.java

```diff
@@ -356,16 +356,19 @@ private void validateRequiredFields(LinkedHashMap<String, Boolean> siblings, Xml
 
         for (String key : xmlParserData.fieldHierarchy.peek().keySet()) {
             // Validate required array size
-            if (xmlParserData.fieldHierarchy.peek().get(key).getFieldType().getTag() == TypeTags.ARRAY_TAG) {
```

Comment:
The mistake I made previously is that I used `key` which is modified name but here we have to use the original fieldName.